packages/nimble-components/src/select/index.ts

```diff
@@ -376,6 +396,11 @@ export class Select
             case 'hidden': {
                 if (isNimbleListOption(sourceElement)) {
                     sourceElement.visuallyHidden = sourceElement.hidden;
+                } else if (isNimbleListOptionGroup(sourceElement)) {
+                    sourceElement.listOptions.forEach(e => {
```

Comment:
I think this is the only code that updates `visuallyHidden` on child list options when a group is hidden. If so, you should make sure everything works correctly when a group that is already hidden is slotted into the component. Since this is a change handler, I wouldn't expect this code path to be hit in that case. I suspect that may be problematic.

Reply:
Line 1093 (the `getGroupedOptions` implementation) handles this case. I don't think I have test coverage for this though so I'll add it.